pkg/queryfrontend/config.go

```diff
@@ -242,14 +245,25 @@ type LabelsConfig struct {
 // Validate a fully initialized config.
 func (cfg *Config) Validate() error {
 	if cfg.QueryRangeConfig.ResultsCacheConfig != nil {
-		if cfg.QueryRangeConfig.SplitQueriesByInterval <= 0 {
-			return errors.New("split queries interval should be greater than 0 when caching is enabled")
+		if cfg.QueryRangeConfig.SplitQueriesByInterval <= 0 && !cfg.isDynamicSplitSet() {
```

Comment:
This change might be out of scope for this PR, but I wonder why we even have this constraint. @yeya24 any ideas?

Reply:
I think this was inherited from Cortex (only by looking at git history). https://github.com/thanos-io/thanos/commit/17343c16

Reply:
Right, we don't allow negative split intervals.